components/suggest/src/suggestion.rs

```diff
@@ -31,3 +46,184 @@ impl Suggestion {
         matches!(self, Self::Amp { .. })
     }
 }
+
+/// Replaces all template parameters in a "raw" sponsored suggestion URL,
+/// producing a "cooked" URL with real values.
+pub(crate) fn cook_raw_suggestion_url(raw_url: &str) -> String {
+    let now = Local::now();
+    let mut cooked_url = String::new();
+    let mut last_index = 0;
+    for (index, _) in raw_url.match_indices(TIMESTAMP_TEMPLATE) {
```

Comment:
Replacing just the first occurrence sounds good to me!

> If the partner breaks that contract by including more than one template, we should reject it at the ingestion phase and let them fix that. WDYT?

Agreed—just to make sure we're on the same page, though, I think it would be valuable to do this when we ingest into Remote Settings, not when we ingest from Remote Settings into our SQLite database.

Reply:
Yes, I meant enforcing that at the ingestion phase before loading data into Remote Settings. Oh, right. We also have another ingestion here from RS to SQLite, but that's not what was referring to, sorry about that. 😅

Reply:
No worries, I figured that's what you meant, and wanted to double-check!